### PR TITLE
Campaign batch mongodb

### PR DESCRIPTION
**Pull request by vlimant**
vlimant@vocms0275.cern.ch >python campaignsConfiguration.py --help
Usage: campaignsConfiguration.py [options]

Options:
  -h, --help            show this help message and exit
  --dump=DUMP           dump the whole content in this file
  --load=LOAD           synchronize the db with the content of this file
  -n NAME, --name=NAME  a campaign name to be viewed/added/updated
  --remove              remove the specified campaign
  -c CONFIGURATION, --configuration=CONFIGURATION
                        a json doc for the campaign to be add/updated
  -p PARAMETER, --parameter=PARAMETER
                        a single parameter to be updated of the form key:value
                        or a.b.key:value for nested
can be used to update the campaign configuration, from a full json, or a single json for a new campaign, or updating one parameter only

### CODE REVIEW

**Comment by vlimant** [comment]
vlimant@vocms0275.cern.ch >python campaignsConfiguration.py --dump campaigns.json

may be used to update the json that is put in github, in case people deem this necessary

**Comment by vlimant** [comment]
this removes another string to eos @SvenjaPflitsch if you have time to go and do the test, we can push this in

**Comment by vlimant** [comment]
@SvenjaPflitsch let's do this, so that we can put it behind

**Comment by vlimant** [comment]
when you are sufficiently confident that editing campaign configuration is ok, let's move this in

**Comment by SvenjaPflitsch** [comment]
I tested the different options and documented the different examples in the UO twiki under
"Using campaignsConfiguration.py "

Unfortunately at the end of the tests the "dump" option did not work any more, even though I did not change anything in the code that should affect it and it worked in the beginning. 

python campaignsConfiguration.py --dump campaignTest6.json
Traceback (most recent call last):
  File "campaignsConfiguration.py", line 39, in <module>
    uc[content.pop("name")] = content
KeyError: 'name'

File: campaignsConfiguration.py [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
 post = {}
 if options.configuration:
-    post.update(json.loads(options.value))
+    post.update(json.loads(options.configuration))
```
Comment (vlimant) [inline campaignsConfiguration.py]: you would need to add post['name'] = options.name

**Comment by vlimant** [comment]
@SvenjaPflitsch I think https://github.com/CMSCompOps/WmAgentScripts/pull/336/commits/4673a50520bb433f0f7cb93a52b36db441255e9b#r236213838 is the reason.
I would --load campaigns.json and see whether it dumps back properly

**Comment by SvenjaPflitsch** [comment]
I tried --load and afterwards --dump, but the error remains the same.

**Comment by SvenjaPflitsch** [comment]
Thanks, it is working now. I will redo some of the tests just to be sure but then it should be ok. I will need something like 20 min.

**Comment by SvenjaPflitsch** [comment]
Ok everything works without any errors.

**Comment by vlimant** [comment]
can you load the relval in mongodb and I merge

**Comment by SvenjaPflitsch** [comment]
RelVal campaigns have been added.

**Comment by vlimant** [comment]
let's see how things go. you might want put the output of --dump in campaigns.json and push it to github for safety

**Comment by SvenjaPflitsch** [comment]
campaigns.json is up to date. I made a copy and a --dump version to be sure.

**Comment by SvenjaPflitsch** [comment]
It looks like the Autumn18 campaigns were not uploaded to the database using the --load option. I re-checked the campaigns.json file and they are definitely in the one I used for the --load option yesterday.
I tried it again to be sure, but the result is the same. 

However, when I add them one-by-one with the --configuration option, it seems to be alright. They show up with the --name option a well as with the --dump option.

**Pull request merged**